sdk/core/core/src/az_json_parser.c

```diff
@@ -31,7 +32,7 @@ AZ_NODISCARD AZ_INLINE bool az_json_is_white_space(az_result_byte c) {
   return false;
 }
 
-AZ_NODISCARD AZ_INLINE bool az_json_is_e(az_result_byte c) {
+AZ_NODISCARD AZ_INLINE bool az_json_is_e(uint8_t c) {
```

Comment:
Please add a comment as to why this function exists. It's very odd to see a function that looks specifically for 'E'. I think this is for floating point number parsing, right?